Warehouse.Infrastructure/RestClient/RestClient.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Net.Http;
+using System.Text.Json;
+
+namespace Warehouse.Infrastructure.RestClient
+{
+    public class RestClient<TResource, TIdentifier> : IDisposable where TResource : class
```

Comment:
This will be a useful article because if we manage the http clients ourselves there may be problems.

https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests